logstash-core/lib/logstash/environment.rb

```diff
@@ -1,7 +1,38 @@
 # encoding: utf-8
 require "logstash/errors"
+require "logstash/config/cpu_core_strategy"
+require "logstash/settings"
 
 module LogStash
+
+  [
+                  Setting.new("node.name", String, Socket.gethostname),
+           BooleanSetting.new("config.allow_env", false),
+  ExistingFilePathSetting.new("config.path", nil, false),
+                  Setting.new("config.string", String, nil, false),
+           BooleanSetting.new("config.test", false),
+           BooleanSetting.new("config.auto_reload", false),
+                  Setting.new("config.reload_interval", Numeric, 3),
+           BooleanSetting.new("metric.collect", true) {|v| v == true }, # metric collection cannot be disabled
+                  Setting.new("pipeline.id", String, "main"),
+                  Setting.new("pipeline.workers", Numeric, LogStash::Config::CpuCoreStrategy.maximum),
+                  Setting.new("pipeline.output.workers", Numeric, 1),
+                  Setting.new("pipeline.batch.size", Numeric, 125),
+                  Setting.new("pipeline.batch.delay", Numeric, 5), # in milliseconds
+           BooleanSetting.new("pipeline.unsafe_shutdown", false),
+                  Setting.new("plugin.paths", Array, []),
+                  Setting.new("ruby_shell", String, nil, false),
+           BooleanSetting.new("debug", false),
+           BooleanSetting.new("debug.config", false),
+           BooleanSetting.new("verbose", false),
+           BooleanSetting.new("quiet", false),
+           BooleanSetting.new("version", false),
+           BooleanSetting.new("help", false),
+                  Setting.new("log.path", String, nil, false),
+                  Setting.new("web_api.http.host", String, "127.0.0.1"),
+                  Setting.new("web_api.http.port", Numeric, 9600),
+  ].each {|setting| SETTINGS.register(setting) }
```

Comment:
I am wondering if the `BooleanSetttings` should be defined in the Setting class, this would make the Settings acts as a module.

```ruby
class Setting
  attr_reader :name

  def initialize(name)
    @name = name
  end

  class Boolean < Setting
    def initialize(name)
      @name = "boolean-#{name}"
    end
  end
end

puts Setting.new("Bojourno").name
puts Setting::Boolean.new("adios").name
```